### PR TITLE
Fields for determining OCW URLs

### LINKED ISSUE #3144: Add all attributes required to construct an OCW URL (department slug, course slug, etc) to models, index
Include the parts necessary for the OCW URL construction (course slug, page slug, etc) - current and future - in the index and the model.  These parts will be used to generate the full URL for `Take this Course` link on the front end (#3133)

Current sample URL structure: 
`/courses/electrical-engineering-and-computer-science/6-331-advanced-circuit-techniques-spring-2002/download-course-materials`

department slug: `electrical-engineering-and-computer-science`
course slug: `6-331-advanced-circuit-techniques-spring-2002`
page: `download-course-materials`

New sample URL structure:
`/courses/6-331-advanced-circuit-techniques-spring-2002/sections/download-course-materials`

(department slug removed, `sections` added)

### PR DESCRIPTION
**Pull request by mbertrand**
Depends on #3141

#### Pre-Flight checklist
- [x] Migrations
  - [x] Migration is backwards-compatible with current production code
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #3144 

#### What's this PR do?
Adds additional fields to the LearningResourceRun & ContentFile models and ES index that can be used to construct OCW URL's.

#### How should this be manually tested?
- Recreate the index
- Run `backpopulate_ocw_files` (assuming you already imported OCW courses, if not run `backpopulate_ocw_data --overwrite --skip-s3`)
- Check that the ES index contains fields for `department_slug` under each run.
- Run this search via curl/postman for resources.  Each resource should include a `section_slug` field (may be null), and a `run_department_slug` field.

```
{  "size": 100,
    "post_filter": {
        "bool": {
            "must": [
                {
                    "bool": {
                        "should": [
                            {
                                "term": {
                                    "object_type.keyword": "resourcefile"
                                }
                            }
                        ]
                    }
                },
                {
                    "bool": {
                        "should": [
                            {
                                "has_parent": {
                                    "parent_type": "resource",
                                    "query": {
                                        "bool": {
                                            "should": [
                                                {
                                                    "term": {
                                                        "offered_by": "OCW"
                                                    }
                                                }
                                            ]
                                        }
                                    }
                                }
                            }
                        ]
                    }
                }
            ]
        }
    }
}
```

### CODE REVIEW

File: course_catalog/serializers.py [hunk 1]
```diff
@@ -297,6 +297,11 @@ class CourseSerializer(BaseCourseSerializer, LearningResourceRunMixin):
     object_type = serializers.CharField(read_only=True, default="course")
     audience = serializers.ReadOnlyField()
     certification = serializers.ReadOnlyField()
+    department_slug = serializers.SerializerMethodField()
+
+    def get_department_slug(self, instance):
```
Comment (abeglova) [inline course_catalog/serializers.py:302]: You can just replace line 300 with `department_slug =  serializers.ReadOnlyField()` similar to audience and certification and remove `get_department_slug`

**Review by abeglova — CHANGES_REQUESTED** [review]
Works as advertised. Back to you with a minor comment

**Review by abeglova — APPROVED** [review]

File: course_catalog/serializers.py (hunk 1 above)
Comment (mbertrand) [reply course_catalog/serializers.py]: Thank! Changed it.

**Comment by abeglova** [comment]
👍

**Pull request merged**